Social: Update post publish review prompt

manzoorwanijk opened the pull request:
Closes SOCIAL-249

Our review prompt in the sidebar looks a bit off and doesn't follow the core component styles. I updated the component to use the Core Notice component and took the opportunity to clean up other related things.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Update review prompt to use Core notice component instead of our own custom one.
* Convert `useJetpackSocialPreferences` to a generic user preferences hook, making it easier to add more preferences.
* Remove some unused code from Social plugins and Publicize package for dismissing notices.
* Updated the review prompt logic to use the core preferences store to save review prompt dismissed preference instead of a custom REST endpoint.
* Updated the logic to allow reviews for both Social and Jetpack plugins.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Publish a post and make sure that it's set to be shared to at least one social media account.
* On the post-publish screen, confirm that you see the review prompt.
* Confirm that the panel has the title "Jetpack Social"
* Open the "Leave review" link in a new tab.
* Confirm that it opens the Jetpack reviews or Social reviews page depending up whether the Social plugin is active.
* Click on "Dismiss"
* Confirm that the review prompt is hidden immediately.
* Now publish another post
* Confirm that the review prompt is not shown anymore. 
* If you need to do more testing by resetting the preference, you can run this in the browser console.

```js
wp.data.dispatch('core/preferences').toggle('jetpack/social', 'review_prompt_dismissed')
```


| Before | After |
|--------|--------|
| <img width="278" height="407" alt="Screenshot 2025-12-23 at 2 36 21 PM" src="https://github.com/user-attachments/assets/16c950eb-9ca3-4f3a-a738-8afccc76a26b" /> |  <img width="283" height="395" alt="Screenshot 2025-12-23 at 2 45 58 PM" src="https://github.com/user-attachments/assets/b86830a0-ad97-45b7-a5fa-682786e5101e" /> |

manzoorwanijk committed:
Remove unused Dismissed Notices handler and tests

manzoorwanijk committed:
Remove unused pricing page option and method

manzoorwanijk committed:
Remove the unused dismiss review endpoint

manzoorwanijk committed:
Update review prompt to use Core notice component

manzoorwanijk committed:
Convert useJetpackSocialPreferences to a generic user preferences hook

manzoorwanijk committed:
Use preferences store to save review prompt dismissed

manzoorwanijk committed:
Allow reviews for both Social and Jetpack plugins

manzoorwanijk committed:
Add changelog

manzoorwanijk committed:
Add title and logo to the panel

manzoorwanijk committed:
Show review prompt only for self-hosted sites

manzoorwanijk commented on projects/js-packages/publicize-components/src/hooks/use-social-user-preferences/index.ts:
```diff
@@ -0,0 +1,73 @@
+type Preference = keyof SocialUserPreferences;
+
+const PREFERENCES: Record< Preference, string > = {
+	prePublishConfirmation: 'pre_publish_confirmation',
+	reviewPromptDismissed: 'review_prompt_dismissed',
```
That is fine. It has not been released yet.

manzoorwanijk committed:
Add tests for useSocialUserPreferences hook

manzoorwanijk committed:
Merge branch 'trunk' into update/social/review-prompt

haqadn commented on projects/js-packages/publicize-components/src/components/block-editor/components/post-publish-panels.tsx:
```diff
@@ -1,14 +1,15 @@
 const PostPublishPanels = () => {
 	return (
 		<ThemeProvider targetDom={ document.body }>
+			{ isJetpackSelfHostedSite() && <PostPublishReviewPrompt /> }
```
Didn't notice it at first, and was wondering why the review prompt isn't appearing on my JN site. I am curious about why we want to hide it for hosted sites. Don't we want to show it in Pressable, Bluehost or even WPCOM sites? I understand we might want to hide it because JP comes prepackaged. I just want to understand the rationale clearly.

manzoorwanijk replied:
Before this PR, the review prompt was shown only if the Social standalone plugin was active. In this PR, I [changed the logic](https://github.com/Automattic/jetpack/pull/46398/commits/d67e40a609511d722b91d67a818b6acc3d0d596d) to allow it for the Jetpack plugin as well, and why not? So, I thought to add a guardrail to ensure it's shown on the self-hosted sites only. Maybe a better solution/fix is to skip it only on WPCOM? What do you think?

haqadn replied:
I don't have a strong opinion on this. Not showing where it's prepackaged makes sense. But, also it is a distinct product and the notice is not too intrusive. I'm happy either way.

robertsreberski replied:
Do we have data on the click rate of this review prompt?

I believe it's better to display it to more users, especially since it's not intrusive. It's more of "how did we do?". I would say we could display it even on wpcom.

manzoorwanijk replied:
I could not find any data for the event. May be nobody clicked on it. 😄 

Yeah, we can show it on all sites but I am not sure of wpcom. May be let us start with all sites, except wpcom.

manzoorwanijk replied:
Enabled it for all sites, except wpcom - https://github.com/Automattic/jetpack/pull/46398/commits/d35d752a3fbda21194ded9cf5ffc5e7b802017f1

haqadn commented on projects/js-packages/publicize-components/src/components/review-prompt/index.jsx:
```diff
@@ -16,25 +16,37 @@ const ReviewPrompt = ( { href, onClose } ) => {
+			<Notice
+				isDismissible={ false }
+				status="info"
+				actions={ [
+					{
+						variant: 'primary',
+						label: __( 'Leave a Review', 'jetpack-publicize-components' ),
+						url: href,
+						className: 'is-compact',
+					},
```
> The tracking event for clicking the "Leave a Review" link has been removed. While the dismiss click tracking remains, the click tracking for the primary action (leaving a review) is missing. This makes it harder to measure the effectiveness of this prompt and understand user engagement with the review request.

I assume this is because we will see the stats on jp redirect for this?

manzoorwanijk replied:
Yes. Also, the Notice component actions have the limitations right now that we can't use `onClick` with `url`

More details in https://github.com/WordPress/gutenberg/issues/74090

haqadn reviewed (changes requested):
I'm not seeing the margin between the buttons.

<img width="282" height="354" alt="Google Chrome 2025-12-23 13 44 37" src="https://github.com/user-attachments/assets/724df8e3-ef4b-4d09-a67a-5723582891a8" />

haqadn reviewed (dismissed):
I'm not seeing the margin between the buttons.

<img width="282" height="354" alt="Google Chrome 2025-12-23 13 44 37" src="https://github.com/user-attachments/assets/724df8e3-ef4b-4d09-a67a-5723582891a8" />

manzoorwanijk committed:
Add gap between buttons

manzoorwanijk commented:
I have added some gap between the buttons for now.

robertsreberski commented on projects/js-packages/publicize-components/src/components/review-prompt/style.module.scss:
```diff
@@ -0,0 +1,6 @@
+.reviewPrompt {
+
+	:global(.components-notice__actions) {
```
Thanks for adding this fallback. Let's also add a short commentary why it's needed, as you explained it in the comment: https://github.com/Automattic/jetpack/pull/46398#issuecomment-3688841905

manzoorwanijk replied:
Done. Thanks

robertsreberski reviewed (dismissed):
Works good for me, and I'm glad for adding the fallback styling here for prior versions of WP.

Let's add a little commentary to future us/other maintainers, on why we decided to add this styling.

Other than this, it's very nice PR, and I appreciate seeing tests added as well!

manzoorwanijk committed:
Add comments

manzoorwanijk committed:
Enable the prompt on all sites, except wpcom

manzoorwanijk committed:
Oops! Update import path

manzoorwanijk committed:
Add TODO for tracking on review prompt link

Added a TODO comment to restore tracking on the 'Leave a Review' link once Notice actions support onClick for links. Reference to related Gutenberg PR included.

manzoorwanijk commented on projects/js-packages/publicize-components/src/components/review-prompt/index.jsx:
```diff
@@ -16,25 +17,38 @@ const ReviewPrompt = ( { href, onClose } ) => {
+				status="info"
+				className={ style.reviewPrompt }
+				actions={ [
+					{
+						variant: 'primary',
+						label: __( 'Leave a Review', 'jetpack-publicize-components' ),
+						url: href,
+						className: 'is-compact',
```
Added a TODO comment

manzoorwanijk commented on projects/js-packages/publicize-components/src/components/post-publish-review-prompt/index.tsx:
```diff
@@ -27,28 +27,29 @@ const PostPublishReviewPrompt = () => {
 		return null;
 	}
 
+	const pluginReviewUrl = getRedirectUrl(
+		// If the social plugin is active, direct to that review page
+		getSocialScriptData().plugin_info.social.version
```
It's null if the plugin is not installed.

manzoorwanijk committed:
Add back tracking

manzoorwanijk commented on projects/js-packages/publicize-components/src/components/review-prompt/index.jsx:
```diff
@@ -27,14 +28,32 @@ const ReviewPrompt = ( { href, onClose } ) => {
+						label: __( 'Leave a Review', 'jetpack-publicize-components' ),
+						url: href,
+						className: 'is-compact',
+						onClick: recordReviewClick,
```
https://github.com/WordPress/gutenberg/pull/74094 is now merged. So, I have added back the `onClick` event here alongside `url`.

grzegorz-cp reviewed (approved):
Good changes 👍
